### DISCUSSION TITLE
Violation error when running CoreWebView2.ExecuteScriptAsync(strJavascript);

### DISCUSSION THREAD

**Discussion by cyberfizz** [MicrosoftEdge/WebView2Feedback#3889] [Q&A]
We run a script to automate entering of text and logins when the page is loaded. We have have an event for when NavigationCompleted which enters the javascript, but in the developer console it looks to be blocked:

![image](https://user-images.githubusercontent.com/7288622/124136803-008deb00-da7d-11eb-8b90-e8590220e7d5.png)

This works fine using the same method in cefsharp for instance, using the same methods, but seems to fail with WebView2. 

If i try the commands direct in to the console they work fine, only when done via the ExecuteScript function.

**Comment by champnic** [comment]
Hey @cyberfizz - Are you able to share the script that's causing problems for you? Looks like Chromium automatically blocks document.write("<script ...>") calls as a perf improvement. But I'm surprised it would work directly in the console in that case. Can you avoid those calls? If not, we may need to make some changes to allow these calls (or add a setting to allow these calls, etc.).

**Comment by champnic** [comment]
This document might also be helpful in potential solutions: https://developers.google.com/web/updates/2016/08/removing-document-write
